test/e2e/README.md

```diff
@@ -0,0 +1,346 @@
+    sc := scenarioCtx(ctx)
+    // Implementation
+    return nil
+}
+```
+
+### 3. Use Existing Steps
+
+Leverage existing steps for common operations:
+
+- **Setup**: `Given OLM is available`, `And "test" catalog serves bundles`
+- **Resource Management**: `When ClusterExtension is applied`, `And resource is applied`
+- **Assertions**: `Then ClusterExtension is available`, `And bundle "..." is installed`
+- **Conditions**: `Then ClusterExtension reports Progressing as True with Reason Retrying:`
+
+### 4. Variable Substitution
+
+Use these variables in YAML templates:
+
+- `$NAME`: Scenario-specific ClusterExtension name (e.g., `ce-123`)
+- `$TEST_NAMESPACE`: Scenario-specific namespace (e.g., `ns-123`)
+
+### 5. Feature Tags
+
+Use tags to conditionally run scenarios based on feature gates:
+
+```gherkin
+@WebhookProviderCertManager
+Scenario: Install operator having webhooks
+```
+
+Scenarios are skipped if the feature gate is not enabled on the deployed controller.
+
+## Running Tests
+
+### Run All Tests
+
+```bash
+make test-e2e
+```
+
+or
+
+```bash
+make test-experimental-e2e
+```
+
+
+### Run Specific Feature
+
+```bash
+go test test/e2e/features_test.go -- features/install.feature
+```
+
+### Run Specific Scenario by Tag
+
+```bash
+go test test/e2e/features_test.go --godog.tags="@WebhookProviderCertManager"
+```
+
+### Run with Debug Logging
+
+```bash
+go test -v test/e2e/features_test.go --log.debug
+```
+
+### CLI Options
+
+Godog options can be passed after `--`:
+
+```bash
+go test test/e2e/features_test.go \
+  --godog.format=pretty \
+  --godog.tags="@WebhookProviderCertManager"
+```
+
+Available formats: `pretty`, `cucumber`, `progress`, `junit`
+
+**Custom Flags:**
+
+- `--log.debug`: Enable debug logging (development mode)
+- `--k8s.cli=<path>`: Specify path to Kubernetes CLI (default: `kubectl`)
+  - Useful for using `oc` or a specific kubectl binary
+
+**Example:**
+
+```bash
+go test test/e2e/features_test.go --log.debug --k8s.cli=oc
+```
+
+### Environment Variables
+
+- `KUBECONFIG`: Path to kubeconfig file (defaults to `~/.kube/config`)
+- `E2E_SUMMARY_OUTPUT`: Path to write test summary (optional)
+- `LOCAL_REGISTRY_HOST`: Local registry for catalog images
+
+## Design Patterns
+
+### 1. Scenario Isolation
+
+Each scenario runs in its own namespace with unique resource names, ensuring complete isolation:
+
+- Namespace: `ns-{scenario-id}`
+- ClusterExtension: `ce-{scenario-id}`
+
+### 2. Automatic Cleanup
+
+The `ScenarioCleanup` hook ensures all resources are deleted after each scenario:
+
+- Kills background processes (e.g., kubectl port-forward)
+- Deletes ClusterExtensions
+- Deletes namespaces
+- Deletes added resources
+
+### 3. Declarative Resource Management
+
+Resources are managed declaratively using YAML templates embedded in feature files as docstrings:
+
+```gherkin
+When ClusterExtension is applied
+"""
+  apiVersion: olm.operatorframework.io/v1
+  kind: ClusterExtension
+  metadata:
+    name: $NAME
```

Comment:
Nit, here and elsewhere. Since we now support bracket variables, and that's what's used in the `.feature` files.
```suggestion
    name: ${NAME}
```
Although I realize both formats are accepted.